File: Design_Pattern_and_Principles/ProxyPatternExample/RealImage.java
public class RealImage implements Image {
    private String filename;

    public RealImage(String filename) {
        this.filename = filename;
        loadFromDisk();
    }

    private void loadFromDisk() {
        System.out.println("Loading heavy asset file [" + filename + "] from storage...");
    }

    public void display() {
        System.out.println("Displaying " + filename);
    }
}

File: Design_Pattern_and_Principles/ProxyPatternExample/RealImage.cs
using System;

public class RealImage : IImage {
    private string filename;

    public RealImage(string filename) {
        this.filename = filename;
        LoadFromDisk();
    }

    private void LoadFromDisk() {
        Console.WriteLine("Loading heavy asset file [" + filename + "] from storage...");
    }

    public void Display() {
        Console.WriteLine("Displaying " + filename);
    }
}